### ISSUE TITLE
Add support for GB-2312 encoding in subject header

### ISSUE THREAD

**Issue by zmyxingyu** [k9mail/k-9#2499]
### Expected behavior
K-9 should correctly convert multi-encoded-word GB2312 encoded subjects into decoded text.

### Actual behavior
Does not display properly.

### Steps to reproduce
I have received a mail like this
```
Date: Tue, 11 Apr 2017 11:34:38 +0800
MIME-Version: 1.0
Content-Type: multipart/mixed;
	boundary="__=_Part_Boundary_007_006502.007471"
Subject: =?gb2312?B?Obv9t9az6cnu29rHsLqju6rHyLPHSlfN8rrAvsa16qOsuPzT0DIwvNIzOTnU?=
	 =?gb2312?B?qr6r0aG439DHytTLr77Gteq1yMTjwLSjoaOoQUSjqQ?=
```
this application will not display properly
but  i can normal reading on gmail

### Environment
K-9 Mail version:5.206

Android version:

Account type (IMAP, POP3, WebDAV/Exchange):

**Comment by zmyxingyu** [comment]
In fact this is a bug,  DecoderUtil.decodeEncodedWords() does not solve some middle with encoded string .
as  =?utf-8?B?5Liq5Lq66YKu566xOkJVRyAjMzAyNDY6OumCruS7tuato+aWh+mZhOS7tuWQ?=
 =?utf-8?B?jeensOecgeeVpeaYvuekuuS8mOWMlg==?=

**Comment by philipwhiuk** [comment]
Is it actually MIME spec compliant to do that?

**Comment by zmyxingyu** [comment]
yes，I have found many applications do the processing.But temporarily haven't found a source algorithm

**Comment by zmyxingyu** [comment]
Remove the middle of the code under a certain algorithm to solve the problem, but mime4j some cases still have this bug

**Comment by philipwhiuk** [comment]
Okay. Per RFC2047 it's definitely legitimate to have multiple 'encoded-word' sections separated by a space.

I'll look into fixing this.

**Comment by zmyxingyu** [comment]
Thank you for your reply and hard work,  good luck

**Comment by philipwhiuk** [comment]
I finally got some time to look into this (or maybe I already did and forgot to respond). The bug definitely isn't in the decoder. I'm committing some extra tests that show this.

Given it wraps onto multiple lines I'm wondering if this is a header decoding issue.... I don't have a test email to try this though - if you could send one to philip@whiuk.com then I can debug it...

**Comment by zmyxingyu** [comment]
I can't find the email.Specific questions are some client code is not standard.GB2312 encoding, for example, Chinese accounting for 2 bytes, some string coding algorithm is divided into two pieces of code, if just separate the characters of the last two bytes, will lead to parse the code decoding algorithm.But some algorithm is compatible with the code, such as gmail

**Comment by zmyxingyu** [comment]
![image](https://user-images.githubusercontent.com/11452338/29965867-02ee748a-8f42-11e7-8c6d-8e3cae877b84.png)
Subject: =?gb2312?B?Obv9t9az6cnu29rHsLqju6rHyLPHSlfN8rrAvsa16qOsuPzT0DIwvNIzOTnU?=
	 =?gb2312?B?qr6r0aG439DHytTLr77Gteq1yMTjwLSjoaOoQUSjqQ?=
the expect words on the picture

**Comment by philipwhiuk** [comment]
It's being split in the middle of a Chinese character. Which is bad behaviour by the sender's email client. But I've got a PR that handles it - #2722

**Comment by philipwhiuk** [comment]
Fixed in #2725
